src/ring_theory/dedekind_domain.lean

```diff
@@ -752,6 +752,23 @@ theorem ideal.prime_iff_is_prime {P : ideal A} (hP : P ≠ ⊥) :
   prime P ↔ is_prime P :=
 ⟨ideal.is_prime_of_prime, ideal.prime_of_is_prime hP⟩
 
+lemma ideal.pow_succ_lt (I : ideal A) (hI0 : I ≠ ⊥) (hI1 : I ≠ ⊤) (e : ℕ) : I^(e+1) < I^e :=
+ideal.dvd_not_unit_iff_lt.mp
+⟨pow_ne_zero _ hI0, I, mt is_unit_iff.mp hI1, pow_succ' I e⟩
+
+lemma ideal.pow_lt_self (I : ideal A) (hI0 : I ≠ ⊥) (hI1 : I ≠ ⊤) (e : ℕ) (he : 2 ≤ e) : I^e < I :=
+begin
+  induction e with e ih, { norm_num at he },
+  cases he.eq_or_lt with he he,
+  { simpa [← nat.succ_injective he] using I.pow_succ_lt hI0 hI1 1 },
+  exact (I.pow_succ_lt hI0 hI1 e).trans (ih (nat.succ_le_succ_iff.mp he))
+
+end
```

Comment:
This should follow immediately after rewriting `I` to `I^1` and getting hold of `1 < e` in whatever the obvious way is.